client/src/test/java/io/pravega/client/netty/impl/ConnectionPoolingTest.java

```diff
@@ -0,0 +1,282 @@
+
+        @Override
+        public void channelRead(ChannelHandlerContext ctx, Object message) {
+            if (message instanceof WireCommands.Hello) {
+                ctx.write(message);
+                ctx.flush();
+            } else if (message instanceof WireCommands.ReadSegment) {
+                WireCommands.ReadSegment msg = (WireCommands.ReadSegment) message;
+                ctx.write(readResponseGenerator.apply(msg.getRequestId()));
+                ctx.flush();
+            }
+        }
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        // Configure SSL.
+        port = TestUtils.getAvailableListenPort();
+        final SslContext sslCtx;
+        if (ssl) {
+            try {
+                sslCtx = SslContextBuilder.forServer(new File("../config/cert.pem"), new File("../config/key.pem")).build();
+            } catch (SSLException e) {
+                throw new RuntimeException(e);
+            }
+        } else {
+            sslCtx = null;
+        }
+        boolean nio = false;
+        EventLoopGroup bossGroup;
+        EventLoopGroup workerGroup;
+        try {
+            bossGroup = new EpollEventLoopGroup(1);
+            workerGroup = new EpollEventLoopGroup();
+        } catch (ExceptionInInitializerError | UnsatisfiedLinkError | NoClassDefFoundError e) {
+            nio = true;
+            bossGroup = new NioEventLoopGroup(1);
+            workerGroup = new NioEventLoopGroup();
+        }
+
+        ServerBootstrap b = new ServerBootstrap();
+        b.group(bossGroup, workerGroup)
+         .channel(nio ? NioServerSocketChannel.class : EpollServerSocketChannel.class)
+         .option(ChannelOption.SO_BACKLOG, 100)
+         .handler(new LoggingHandler(LogLevel.INFO))
+         .childHandler(new ChannelInitializer<SocketChannel>() {
+             @Override
+             public void initChannel(SocketChannel ch) throws Exception {
+                 ChannelPipeline p = ch.pipeline();
+                 if (sslCtx != null) {
+                     SslHandler handler = sslCtx.newHandler(ch.alloc());
+                     SSLEngine sslEngine = handler.engine();
+                     SSLParameters sslParameters = sslEngine.getSSLParameters();
+                     sslParameters.setEndpointIdentificationAlgorithm("LDAPS");
+                     sslEngine.setSSLParameters(sslParameters);
+                     p.addLast(handler);
+                 }
+                 p.addLast(new CommandEncoder(null),
+                           new LengthFieldBasedFrameDecoder(MAX_WIRECOMMAND_SIZE, 4, 4),
+                           new CommandDecoder(),
+                           new EchoServerHandler());
+             }
+         });
+
+        // Start the server.
+        serverChannel = b.bind("localhost", port).awaitUninterruptibly().channel();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        serverChannel.close();
+        serverChannel.closeFuture();
+    }
+
+    @Test
```

Comment:
Missing timeouts (I haven't seen a global one either).

Reply:
added it.